traffic_ops/client/serviceCategory.go

```
@@ -0,0 +1,133 @@
+
+	var data tc.ServiceCategoriesResponse
+	err = json.NewDecoder(resp.Body).Decode(&data)
+	return data.Response, reqInf, nil
+}
+
+// GetServiceCategoryByID gets a service category by the service category id
+func (to *Session) GetServiceCategoryByID(id int) ([]tc.ServiceCategory, ReqInf, error) {
+	route := fmt.Sprintf("%s?id=%d", API_SERVICE_CATEGORIES, id)
+	resp, remoteAddr, err := to.request(http.MethodGet, route, nil)
+	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
+	if err != nil {
+		return nil, reqInf, err
+	}
+	defer resp.Body.Close()
+
+	var data tc.ServiceCategoriesResponse
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, reqInf, err
+	}
+
+	return data.Response, reqInf, nil
+}
+
+// GetServiceCategoryByName gets a service category by the service category name
+func (to *Session) GetServiceCategoryByName(name string) ([]tc.ServiceCategory, ReqInf, error) {
+	url := fmt.Sprintf("%s?name=%s", API_SERVICE_CATEGORIES, url.QueryEscape(name))
+	resp, remoteAddr, err := to.request(http.MethodGet, url, nil)
+	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
+	if err != nil {
+		return nil, reqInf, err
+	}
+	defer resp.Body.Close()
+
+	var data tc.ServiceCategoriesResponse
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return nil, reqInf, err
+	}
+
+	return data.Response, reqInf, nil
+}
+
+// DeleteServiceCategoryByID deletes a service category by service category id
+func (to *Session) DeleteServiceCategoryByID(id int) (tc.Alerts, ReqInf, error) {
+	route := fmt.Sprintf("%s/%d", API_SERVICE_CATEGORIES, id)
+	resp, remoteAddr, err := to.request(http.MethodDelete, route, nil)
+	reqInf := ReqInf{CacheHitStatus: CacheHitStatusMiss, RemoteAddr: remoteAddr}
+	if err != nil {
+		return tc.Alerts{}, reqInf, err
+	}
+	defer resp.Body.Close()
+	var alerts tc.Alerts
+	err = json.NewDecoder(resp.Body).Decode(&alerts)
+	return alerts, reqInf, nil
```

Same as above RE: returning vs ignoring errors